db/migrate/20240603134957_add_group_constraints_by_name_and_organisation.rb

```diff
@@ -0,0 +1,5 @@
+class AddGroupConstraintsByNameAndOrganisation < ActiveRecord::Migration[7.1]
+  def change
+    add_index(:groups, %i[name organisation_id], unique: true)
```

Comment:
yes, this is a good shout and we should have put this in the description.

I checked dev, staging and prod to make sure this wouldn't fail in an awkward way. None of the environments have groups which would cause issues.

One reason it makes sense to get this in now rather than later is to stop this happening, already there are lots of "test groups" with very similar names.

Reply:
Nice! Yeah, makes sense to get this in now